ompi/mca/op/avx/op_avx_component.c

```diff
@@ -0,0 +1,287 @@
+    // OS supports extended processor state management ?
+    if ( (abcd[2] & fma_movbe_osxsave_mask) != fma_movbe_osxsave_mask )
+        return 0;
+#endif  /* defined(__APPLE__) */
+
+    run_cpuid( 7, 0, abcd );
+    flags |= (abcd[1] & avx512f_mask)   ? OMPI_OP_AVX_HAS_AVX512F_FLAG  : 0;
+    flags |= (abcd[1] & avx512_bw_mask) ? OMPI_OP_AVX_HAS_AVX512BW_FLAG : 0;
+    flags |= (abcd[1] & avx2_mask)      ? OMPI_OP_AVX_HAS_AVX2_FLAG     : 0;
+    return flags;
+}
+#endif /* non-Intel compiler */
+
+ompi_op_avx_component_t mca_op_avx_component = {
+    {
+        .opc_version = {
+            OMPI_OP_BASE_VERSION_1_0_0,
+
+            .mca_component_name = "avx",
+            MCA_BASE_MAKE_VERSION(component, OMPI_MAJOR_VERSION, OMPI_MINOR_VERSION,
+                                  OMPI_RELEASE_VERSION),
+            .mca_open_component = avx_component_open,
+            .mca_close_component = avx_component_close,
+            .mca_register_component_params = avx_component_register,
+        },
+        .opc_data = {
+            /* The component is checkpoint ready */
+            MCA_BASE_METADATA_PARAM_CHECKPOINT
+        },
+
+        .opc_init_query = avx_component_init_query,
+        .opc_op_query = avx_component_op_query,
+    },
+};
+
+/*
+ * Component open
+ */
+static int avx_component_open(void)
+{
+    mca_op_avx_component.flags = has_intel_AVX_features();
+    /* A first level check to see what level of AVX is available on the
+     * hardware.
+     *
+     * Note that if this function returns non-OMPI_SUCCESS, then this
+     * component won't even be shown in ompi_info output (which is
+     * probably not what you want).
+     */
+    return OMPI_SUCCESS;
+}
+
+/*
+ * Component close
+ */
+static int avx_component_close(void)
+{
+    /* If avx was opened successfully, close it (i.e., release any
+       resources that may have been allocated on this component).
+       Note that _component_close() will always be called at the end
+       of the process, so it may have been after any/all of the other
+       component functions have been invoked (and possibly even after
+       modules have been created and/or destroyed). */
+
+    return OMPI_SUCCESS;
+}
+
+/*
+ * Register MCA params.
+ */
+static int
+avx_component_register(void)
+{
+    int32_t requested_flags;
+    requested_flags = mca_op_avx_component.flags = has_intel_AVX_features();
+    (void) mca_base_component_var_register(&mca_op_avx_component.super.opc_version,
+                                           "support",
+                                           "Level of SSE/MMX/AVX support to be used (combination of processor capabilities as follow SSE 0x01, SSE2 0x02, SSE3 0x04, SSE4.1 0x08, AVX 0x010, AVX2 0x020, AVX512F 0x100, AVX512BW 0x200) capped by the local architecture capabilities",
+                                           MCA_BASE_VAR_TYPE_INT, NULL, 0, 0,
+                                           OPAL_INFO_LVL_9,
```

Comment:
I don't think anybody should touch these. I added the MCA mostly for me, so that I can test all cases. But they could be tuned by a tool, so I'll bump them up to 6.